### DISCUSSION TITLE
Export data to CSV directly from user side

### DISCUSSION THREAD

**Discussion by timchkor** [Budibase/budibase#1665] [Ideas]
Hello,

It'll be great if user can generate the CSV after filter are applied. Maybe it's more approachable if developer can add some steps after 'Execute Query' action. 

Thanks!

**Comment by mjashanks** [comment]
Thanks @timchkor .

Just to clarify - you would like to
1. Fetch data, with a filter supplied by the user
2. Return that data to the app/user
3. Allow the user to export the data to CSV

Does this describe what you would like?

**Reply by timchkor** [reply]
Yes, that's what I am looking for.

**Reply by NicolasFabre81** [reply]
Linked with https://github.com/Budibase/budibase/discussions/9032 : notice that it is necessary to export the whole data, not only the rows displayed in the screen
